source/plot/ShapePlotData2D.ooc

```diff
@@ -0,0 +1,92 @@
+use ooc-collections
+use ooc-math
+use ooc-draw
+import math
+import PlotData2D
+import svg/Shapes
+
+AnyShape: cover {
+	shape: Shape
+	region: FloatBox2D
+	colorBgra: ColorBgra
+	init: func@ (=shape, =region, =colorBgra)
+}
+
+ShapePlotData2D: class extends PlotData2D {
+	_shapes: VectorList<AnyShape>
+
+	init: func (label := "", colorBgra := ColorBgra new()) {
+		super(label, colorBgra)
+		this _shapes = VectorList<AnyShape> new()
+	}
+
+	addRectangle: func (box: FloatBox2D, colorBgra := ColorBgra new()) {
+		this _shapes add(AnyShape new(Shape Rectangle, box, colorBgra))
+	}
+
+	free: override func {
+		this _shapes free()
+		super()
+	}
+
+	getSvg: func (transform: FloatTransform2D) -> String {
+		result := ""
+		if (!this _shapes empty) {
+			noneColor := ColorBgra new(0, 0, 0, 0)
+			for (i in 0 .. this _shapes count) {
+				object := this _shapes[i]
+				pointA := transform * (object region leftTop)
+				pointB := transform * (object region rightBottom)
+				area := FloatBox2D new(pointA, pointB)
+				color := (noneColor == object colorBgra) ? this colorBgra : object colorBgra
+				match (this _shapes[i] shape) {
+					case Shape Rectangle =>
+						result = result & Shapes rect(area left, area top, area width, area height, color)
+					case =>
+						result = result >> ""
+				}
+			}
+		}
+		result
+	}
+
+	getSvgLegend: func (legendCount, fontSize: Int) -> String {
+		result := ""
+		start := FloatPoint2D new(this legendOffset as Float, this legendOffset + (fontSize * legendCount) as Float - (fontSize as Float) / 2.0f)
+		size := (fontSize as Float) * 0.8f
+		halfLineHeight := (fontSize as Float) / 2.0f
```

Comment:
You could avoid casts if you write it like
```
size := 0.8f * fontSize
halfLineHeight := 0.5f * fontSize
```
But I don't know if you should. @marcusnaslund ?